### ISSUE TITLE
Packet intercepting and sending API

### ISSUE THREAD

**Issue by KosmX** [GeyserMC/Geyser#3365]
### What feature do you want to see added?

API to intercept java/bedrock packets.

I need this API so I can forward bedrock emote messages to JAVA server for [Emotecraft](https://github.com/KosmX/emotes) mod.
And to translate Emotecraft messages to bedrock players (I know about bedrock emote limitation, I already have working workarounds).

### Are there any alternatives?

Modify geyser to forward bedrock emote packets on `geyser:emote` plugin channel, PR was not merged.  

I can mixin Geyser directly on Fabric servers, but on paper or velocity, it is not possible.

**Comment by Camotoy** [comment]
We aren't really interested in promoting an API for packets on both ends, Java and Bedrock. (It may inevitably happen in some form but we wouldn't directly expose MCProtocolLib and Cloudburst Protocol into the API)

In your case: there will definitely be an event fired when a Bedrock player emotes. And I'd rather add a `GeyserConnection#displayEmoteOnPlayer` method.

**Comment by KosmX** [comment]
That could work too.
Last time I asked for this, I was told, there will be an API for packets...

**Comment by Camotoy** [comment]
Yeah, there probably will be eventually (Floodgate already has one in a limited form). Geyser's will certainly not be pretty.
